### PR TITLE
REST endpoint to create a docker container

### PR DESCRIPTION
**Pull request by apache8080**
The endpoint creates the container and returns the url for the jupyter notebook. It also assigns a different port for each container it creates. Currently this only works for acmuiuc images.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM python:2-alpine
```
Comment (narendasan) [inline Dockerfile]: Do we use the dockerfile?

Also Python2?

Comment (apache8080) [reply Dockerfile]: Forgot to remove the Dockerfile. Latest commit has it removed.

File: app.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+app = Flask(__name__)
+CORS(app)
+PORT=5656
+docker_client = cwd.CWDockerClient()
+used_ports = 8889
+
+@app.route('/create_container', methods=['POST'])
+def update_acm_request():
+    global used_ports
+    if not request.json or 'image' not in request.json:
```
Comment (bcongdon) [inline app.py]: Might be good to have a friendlier response as to why the request is rejected

File: app.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        abort(400)
+    print(request.json['image'])
+    print(used_ports)
+    port = used_ports + 1
+    used_ports = port
+    print(used_ports)
+    c_id = docker_client.create_container('', image = request.json['image'], is_gpu = True, port = port)
+    token = docker_client.run_cmd(c_id, 'python ../jupyter_get.py')
+    
+    url = "http://vault.acm.illinois.edu:"+str(port)+"/?token="+str(token)
```
Comment (bcongdon) [inline app.py]: `"http://vault.acm.illinois.edu:{}/?token={}".format(port, token)` is cleaner IMO

File: app.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+from flask_cors import CORS
+import logging
+from cw_libs import clearwaters_docker as cwd
+import commands
+
+app = Flask(__name__)
+CORS(app)
+PORT=5656
+docker_client = cwd.CWDockerClient()
+used_ports = 8889
```
Comment (bcongdon) [inline app.py]: Might be better to use a Flask application global rather than a global variable

http://flask.pocoo.org/docs/0.12/api/#application-globals

Comment (bcongdon) [reply app.py]: More generally, I'm a bit concerned that we're not persisting `used_ports` somewhere (or getting an allocated port map from Docker or some other source of truth). If the server restarts, presumably there may be collisions if one tries to create a container on a previously allocated port.

**Comment by apache8080** [comment]
I have cleaned up the code to now use a sqlite database as the way to keep track of used ports.

**Comment by apache8080** [comment]
This is should be good to merge now.

**Pull request merged**